lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -200,9 +200,11 @@ public function fetch($fetchMode = null)
         $fetchMode = $fetchMode ?: $this->_defaultFetchMode;
         switch ($fetchMode) {
             case \PDO::FETCH_BOTH:
-                return db2_fetch_both($this->_stmt);
+                $result = @db2_fetch_both($this->_stmt);
```

Comment:
The other DB2 operations that emit warnings are already be silenced (db2_execute, db2_prepare, etc.). Same with the Oracle driver.